app/workers/digest_email_generation_worker.rb

```diff
@@ -0,0 +1,39 @@
+class DigestEmailGenerationWorker
+  include Sidekiq::Worker
+
+  def perform(subscriber_id:, digest_run_id:)
+    subscriber = Subscriber.find(subscriber_id)
+    digest_run = DigestRun.find(digest_run_id)
+
+    #get all the content changes that match the subscriber
+    results = SubscriptionContentChangeQuery.call(
+      subscriber: subscriber,
+      digest_run: digest_run
+    )
+
+    email = DigestEmailBuilder.call(
+      subscriber: subscriber,
+      digest_run: digest_run,
+      subscription_content_change_results: results,
+    )
+
+    Email.transaction do
+      email.save!
+
+      subscription_content_changes = results.flat_map do |result|
+        result.content_changes.map do |content_change|
+          {
+            email_id: email.id,
+            subscription_id: result.subscription_id,
+            content_change_id: content_change.id,
+            #TODO we need to set the type on this when there is one
```

Comment:
I guess we don't actually need this since the `type` will be a dynamic scope with a `JOIN`.

Reply:
Fixed in dea0018